src/saml.ts

```diff
@@ -402,10 +402,25 @@ class SAML {
     return buildXmlBuilderObject(request, false);
   }
 
-  _generateLogoutResponse(logoutRequest: Profile) {
+  _generateLogoutResponse(logoutRequest: Profile, success: boolean): string {
     const id = "_" + this._generateUniqueID();
     const instant = this.generateInstant();
 
+    const successStatus = {
+      "samlp:StatusCode": {
+        "@Value": "urn:oasis:names:tc:SAML:2.0:status:Success",
+      },
+    };
+
+    const failStatus = {
+      "samlp:StatusCode": {
+        "@Value": "urn:oasis:names:tc:SAML:2.0:status:Requester",
+        "samlp:StatusCode": {
+          "@Value": "urn:oasis:names:tc:SAML:2.0:status:UnknownPrincipal",
+        },
+      },
+    };
+
```

Comment:
Now that `node-saml/node-saml`'s target is to be standalone SAML library without any relations to any particular nodejs frameworks (unlike `node-saml/passport-saml`) this PR could be merged regardless of what happens to PR https://github.com/node-saml/passport-saml/pull/619

`node-saml` module's client programs should be able to select status codes returned in `LogoutResponse` furthermore client programs should be forced to do so from day one after starting to use this module so that they have to really pay attention to proper `LogoutRequest` processing (of course most of those prorgams would just fallback to provoding status codes indicating `Success` regardless how well `LogoutRequest` handling is implemented but it would be those programs' internal problem/flaw).

So I suggest that change `success` parameter into tuple of statusCodes and merge this PR.
@cjbarth @markstos

Reply:
What do you mean "tuple of statusCodes"? There are lots of status codes to pick from, so I can see how we might want to have arguments that allow a first and second level option. Is that what you mean?

https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-samlpr/96b92662-9bf7-4910-ab16-e1c28bce962b
https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf (Section 3.2.2.2)

Reply:
>… so I can see how we might want to have arguments that allow a first and second level option. Is that what you mean?

Yes

Reply:
I spiked some code to do this and noted as I was looking at all the status codes, that most of them wouldn't apply to this case. There really are only two conditions here the apply: logout success or logout failure. The failure will always be us (the `Responder`, and it will always be of type `UnknownPrincipal`. Do you see another status code that would be appropriate there? If not, it seems wise to keep the API simple and leave this as a `boolean`.

Reply:
@srd90, I was wondering if you had any feedback on my last comment here.

Reply:
@cjbarth If `node-saml` is used independently to implement SP (I am not talking about `passport-saml`) it could be implemented so that if IdP used for some reason e.g. POST binding and if SP is configured/implemented to support LogoutRequest only via redirect binding (or vice versa) it - SP which uses `node-saml` - could respond with `UnsupportedBinding` subcode.

So yes…I see at least one additional code which could be used.